solutions/gold/cf-510C.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-510C
+source: CF
+title: Fox And Names
+author: Daniel Ge
+---
+
+## Solution
+The key idea here is to perform topological sort on the characters of the alphabet. 
+
+We find what must come before what by comparing consecutive strings. If string $name_i$ is lexicographically less than $name_{i+1}$, we compare only the first occurence of a different letter. In other words, we take the common prefix out and compare the first letters. For example, if we are given,
+```
+axcd
+axbe
+```
+We see that the common prefix is "ax", so the first pair of differing letters is 'c' and 'b'. Since "axcd" comes first, that means 'c' comes before 'b' in the desired alphabet.
```

Comment:
> Since "axcd" comes first, that means 'c' comes before 'b' in the desired alphabet.

This doesn't make sense... What do you mean by desired alphabet?

Reply:
the desired alphabet is the answer that we want, do you have any suggestions on how I can rephrase it better? idk how to change it

Reply:
ok